### ISSUE TITLE
A rule asked about a shared run declines in favour of a neighbour whose fix is disabled, and nothing is written

### ISSUE THREAD

**Issue by firefoxic** [stylelint-stylistic/stylelint-stylistic#485]
### What minimal example or steps are needed to reproduce the bug?

```css
a { color:
; }
```

### What minimal configuration is needed to reproduce the bug?

```json
{
	"plugins": [
		"@stylistic/stylelint-plugin"
	],
	"rules": {
		"@stylistic/declaration-colon-space-after": "always",
		"@stylistic/declaration-block-semicolon-space-before": ["never", { "disableFix": true }]
	}
}
```

### What did you expect to happen?

`declaration-colon-space-after` writes its space — `a { color: ; }` — and `declaration-block-semicolon-space-before`, whose fix the configuration turned off, reports the space and leaves it, as it did on `main` at `6c50351`: one run of `--fix`, one warning, and the file at rest.

### What actually happened?

On the branch of #416 nothing is written, and both rules report on every run. `writesSharedRun` names the rule listed last among those speaking of the run its owner, and here that is a rule whose fix is disabled: the colon rule declines in favour of an owner that cannot write, so the run stays as it was.

### Notes

A gap in the reading of a neighbour's setting rather than in either rule: `neighbourSettings` hands over the primary option alone, and `disableFix` stands in the secondary options, which Stylelint reads before it lets a rule's fix run (`lintPostcssResult.mjs`). A neighbour whose fix is disabled still speaks of the run and still reports it, but cannot own it. `whitespaceBeforeSemicolon` (#354) reads its neighbours the same way and writes the whitespace they ask for whether or not their fix is disabled, which does no harm there, since the neighbour asked for it; the reader is one, and the option belongs in it. Found by the review of the branch of #416.
